src/convert.rs

```diff
@@ -534,6 +594,173 @@ pub fn elf_to_tbf<W: Write>(
+        let public_key_path = Path::new(&public_buf);
+
+        let private_key_der = read_rsa_file(private_key_path)
+            .map_err(|e| {
+                panic!(
+                    "Failed to read private key from {:?}: {:?}",
+                    private_key_path, e
+                );
+            })
+            .unwrap();
+
+        let public_key_der = read_rsa_file(public_key_path)
+            .map_err(|e| {
+                panic!(
+                    "Failed to read public key from {:?}: {:?}",
+                    public_key_path, e
+                );
+            })
+            .unwrap();
+
+        let key_pair = signature::RsaKeyPair::from_der(&private_key_der)
+            .map_err(|e| {
+                panic!("RSA4096 could not be parsed: {:?}", e);
+            })
+            .unwrap();
+
+        let public_key = rsa_der::public_key_from_der(&public_key_der);
+
+        let public_modulus = match public_key {
+            Ok((n, _)) => n,
+            Err(_) => {
+                panic!("RSA4096 signature requested but provided public key could not be parsed.");
+            }
+        };
+
+        if key_pair.public_modulus_len() != 512 {
+            // A 4096-bit key should have a 512-byte modulus
+            panic!(
+                "RSA4096 signature requested but key {:?} is not 4096 bits, it is {} bits",
+                private_key_path,
+                private_key_der.len() * 8
+            );
+        }
+        let rng = rand::SystemRandom::new();
+        let mut signature = vec![0; key_pair.public_modulus_len()];
+        let _res = key_pair
+            .sign(
+                &signature::RSA_PKCS1_SHA512,
+                &rng,
+                &output[0..tbfheader.binary_end_offset() as usize],
+                &mut signature,
+            )
+            .map_err(|e| {
+                panic!("Could not generate RSA4096 signature: {:?}", e);
+            });
+        let mut credentials = vec![0; 1024];
+        for i in 0..key_pair.public_modulus_len() {
+            credentials[i] = public_modulus[i];
+        }
+        for i in 0..signature.len() {
+            let index = i + key_pair.public_modulus_len();
+            credentials[index] = signature[i];
+        }
+
+        let rsa4096_credentials = header::TbfFooterCredentials {
+            base: header::TbfHeaderTlv {
+                tipe: header::TbfHeaderTypes::Credentials,
+                length: rsa4096_tlv_len as u16,
+            },
+            format: header::TbfFooterCredentialsType::Rsa4096Key,
+            data: credentials,
+        };
+
+        output.write_all(rsa4096_credentials.generate().unwrap().get_ref())?;
+        footer_space_remaining -= rsa4096_len;
+        if verbose {
+            println!("Added PKCS#1v1.5 RSA4096 signature credential.");
+        }
+    }
+
+    let padding_len = footer_space_remaining;
+    if padding_len >= mem::size_of::<header::TbfHeaderTlv>() {
```

Comment:
I think this check needs to be `if padding_len >= mem::size_of::<header::TbfFooterCredentials>() {`. Otherwise `padding_len` could be 4 yet this would add an 8 byte TLV.